internal/connectors/plugins/public/qonto/accounts.go

```diff
@@ -0,0 +1,104 @@
+package qonto
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/formancehq/go-libs/v3/pointer"
+	"github.com/formancehq/payments/internal/connectors/plugins/currency"
+	"github.com/formancehq/payments/internal/connectors/plugins/public/qonto/client"
+	"time"
+
+	"github.com/formancehq/payments/internal/models"
+)
+
+type accountsState struct {
+	LastUpdatedAt time.Time `json:"lastUpdatedAt"`
+}
+
+func (p *Plugin) fetchNextAccounts(ctx context.Context, req models.FetchNextAccountsRequest) (models.FetchNextAccountsResponse, error) {
+	var oldState accountsState
+	if req.State != nil {
+		if err := json.Unmarshal(req.State, &oldState); err != nil {
+			return models.FetchNextAccountsResponse{}, err
+		}
+	} else {
+		oldState = accountsState{}
+	}
+
+	newState := accountsState{
+		LastUpdatedAt: oldState.LastUpdatedAt,
+	}
+
+	var accounts []models.PSPAccount
+
+	organization, err := p.client.GetOrganization(ctx)
+	if err != nil {
+		return models.FetchNextAccountsResponse{}, err
+	}
+	accounts, newState.LastUpdatedAt, err = fillAccounts(organization.BankAccounts, accounts, oldState)
+	if err != nil {
+		return models.FetchNextAccountsResponse{}, err
+	}
+
+	payload, err := json.Marshal(newState)
+	if err != nil {
+		return models.FetchNextAccountsResponse{}, err
+	}
+
+	/*
+		TODO
+		Need to check the behaviour of the framework:
+		* We are always keeping the "lastUpdatedAt", I believe this gets stored to ensure we don't save/query multiple time
+		the same account -- is that true? Note that the API always return all accounts, so we can't use that for filtering
+		on the API side.
```

Comment:
What you keep in the State depends on the PSP. We often keep the lastUpdatedAt as a paging mechanism for PSPs that don't provide cursors that help us navigate the history. Every time we make a request we want to start again where we left off, and not begin from the beginning of history.